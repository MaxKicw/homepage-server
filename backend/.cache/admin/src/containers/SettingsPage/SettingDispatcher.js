import React, { memo } from 'react';
import { useGlobalContext } from 'strapi-helper-plugin';
import { get } from 'lodash';
import { useParams } from 'react-router-dom';

const SettingDispatcher = () => {
  const { plugins } = useGlobalContext();
  const { pluginId } = useParams();

  const pluginToRender = get(
    plugins,
    [pluginId, 'settings', 'mainComponent'],
    null
  );

  if (!pluginToRender) {
    return null;
  }

  const Compo = pluginToRender;

  return <Compo settingsBaseURL="/settings" />;
};

export default memo(SettingDispatcher);
