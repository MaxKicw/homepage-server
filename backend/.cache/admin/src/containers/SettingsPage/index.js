/**
 *
 * SettingsPage
 *
 */

import React, { memo } from 'react';
import { useGlobalContext, LeftMenu, LeftMenuList } from 'strapi-helper-plugin';
import { get } from 'lodash';
import { Switch, Redirect, Route, useParams } from 'react-router-dom';

import EditView from '../Webhooks/EditView';
import ListView from '../Webhooks/ListView';
import SettingDispatcher from './SettingDispatcher';
import Wrapper from './Wrapper';

function SettingsPage() {
  const { settingId } = useParams();
  const { formatMessage, plugins, settingsBaseURL } = useGlobalContext();

  const pluginsMenu = Object.keys(plugins).reduce((acc, current) => {
    const pluginMenu = get(plugins, [current, 'settings', 'menuSection'], null);

    if (!pluginMenu) {
      return acc;
    }

    acc.push(pluginMenu);

    return acc;
  }, []);

  const menuItems = [
    {
      id: 'global',
      title: { id: 'Settings.global' },
      links: [
        {
          title: formatMessage({ id: 'Settings.webhooks.title' }),
          to: `${settingsBaseURL}/webhooks`,
          name: 'webhooks',
        },
      ],
    },
    ...pluginsMenu,
  ];

  // Redirect to the first static link of the menu
  // This is needed in order to keep the menu highlight
  // The link points to /settings instead of /settings/webhooks
  if (!settingId) {
    return <Redirect to={`${settingsBaseURL}/webhooks`} />;
  }

  return (
    <Wrapper>
      <div className="row">
        <div className="col-md-3">
          <LeftMenu>
            {menuItems.map(item => {
              return <LeftMenuList {...item} key={item.id} />;
            })}
          </LeftMenu>
        </div>
        <div className="col-md-9">
          <Switch>
            <Route
              exact
              path={`${settingsBaseURL}/webhooks`}
              component={ListView}
            />
            <Route
              exact
              path={`${settingsBaseURL}/webhooks/:id`}
              component={EditView}
            />
            <Route
              path={`${settingsBaseURL}/:pluginId`}
              component={SettingDispatcher}
            />
          </Switch>
        </div>
      </div>
    </Wrapper>
  );
}

export default memo(SettingsPage);
