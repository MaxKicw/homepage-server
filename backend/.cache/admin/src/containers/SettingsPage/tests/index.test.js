import React from 'react';
import { Router, Route, Switch } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { GlobalContextProvider } from 'strapi-helper-plugin';
import { IntlProvider } from 'react-intl';

import translationMessages from '../../../translations/en.json';

import SettingsPage from '../index';

const history = createMemoryHistory();

describe('Admin | containers | SettingsPage', () => {
  afterEach(cleanup);

  it('should render SettingsPage', () => {
    const intlProvider = new IntlProvider(
      {
        locale: 'en',
        messages: translationMessages,
      },
      {}
    );
    const { intl: originalIntl } = intlProvider.getChildContext();

    const { asFragment } = render(
      <IntlProvider
        locale="en"
        defaultLocale="en"
        messages={translationMessages}
      >
        <GlobalContextProvider formatMessage={originalIntl.formatMessage}>
          <Router history={history}>
            <Switch>
              <Route>
                <SettingsPage />
              </Route>
            </Switch>
          </Router>
        </GlobalContextProvider>
      </IntlProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
