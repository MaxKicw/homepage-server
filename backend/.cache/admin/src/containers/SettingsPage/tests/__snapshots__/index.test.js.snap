// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin | containers | SettingsPage should render SettingsPage 1`] = `<DocumentFragment />`;
